crates/meroctl/src/common.rs

```diff
@@ -88,3 +88,11 @@ pub enum RequestType {
     Post,
     Delete,
 }
+pub async fn craft_failed_request_message(response: Response, message: &str) -> EyreResult<String> {
+    let status = response.status();
+    let error_text = response.text().await?;
+    Ok(format!(
+        "{} - Status: {}, Error: {}",
+        message, status, error_text
+    ))
+}
```

Comment:
we should define a standardized error structure which will be serialized to JSON

something like

```rust
#[derive(Serialize, Deserialize)]
#[serde(tag = "type", content = "data")]
enum Error<T> {
    ApiError { code: __, data: __ },
    CliError(T),
}
```

Reply:
> plus this would require refactoring how we handle return values (instead of just bailing).

not necessarily.. it can be `bail!(Error::from(response))`

and then at the root, that can be downcasted from `eyre::Report`, passed to `Output`, and then the process exits

> for example, kubectl will ignore the output type if there is an internal error and it will return just the text string

it's fine, just depends on if we want structured errors, cause without it, failures will be more involved to parse from stderr, unless we always want users to "see" exactly what we report here (implying the presence of a terminal).